trustpinning/certs.go

```diff
@@ -266,3 +284,24 @@ func parseAllCerts(signedRoot *data.SignedRoot) (map[string]*x509.Certificate, m
 
 	return leafCerts, intCerts
 }
+
+func checkCertExpiry(cert *x509.Certificate) error {
```

Comment:
Hmm... We seem to have 2 separate sets of checks - these (which check the expiration with no leeway), and the expiration check in `ValidateCertificate`.  Likewise, we check that the RSA key must be at least 2048 in `ValidateCertificate`, but we don't check that for root leaf certificates or intermediate certificates.  For intermediate and root certs, we disallow certs with SHA1.

Is there a reason they should be different?  (might make sense if people are bringing their own CA to maybe be more permissive for the root CA?)

Can we unify the checks into `ValidateCertificate`?  Maybe let that function take a boolean as to whether the expiry check should be performed, etc.?